cmd/dp-dataset-exporter/main.go

```diff
@@ -43,16 +66,16 @@ func main() {
 		cfg.FilterConsumerGroup,
 		kafka.OffsetNewest,
 	)
-	exitIfError(err)
+	updateInitialisation(err, services, serviceConsumer)
```

Comment:
I wonder if having a series of funcs like:
```
 initX() (X, err) {
    // do init
   // update map
}
```
would be cleaner as each init is a single action then. Main then would read like a set of instructions:

```
func main() {
	a, err := initA()
	checkInitErr(err)

	b, err := initB()
	checkInitErr(err)

	b, err := initC()
	checkInitErr(err)
}

func checkInitErr(err error) {
	// log error
	if err != nil {
		//log error
		os.Exit(1)
	}

```
I like having a  reuseable check err/exit func to reduce duplication but I think it should only be called directly from main. All error should be passed back to the top of the app and then main can decide if an exit is necessary.

Reply:
I agree with both points and I will implement `initA()` func. The second point regarding handling of errors for main to decide to exit or not I will also implement it but create a separate function `noExitIfError` as part of this story is to prevent the application exiting when it cannot connect to kafka, mongo, neo4j etc.

Later stories will handle trying to reconnect.